### PR TITLE
feat(rr6): Add shims for React Router 3 → 6

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is tracked by https://github.com/getsentry/frontend-tsc/issues/23

### CODE REVIEW

File: static/app/components/links/link.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import {forwardRef} from 'react';
 import {Link as RouterLink} from 'react-router';
+import {Link as Router6Link} from 'react-router-dom';
 import styled from '@emotion/styled';
 import type {LocationDescriptor} from 'history';
 
+import {USING_REACT_ROUTER_SIX} from 'sentry/constants';
+import {locationDescriptorToTo} from 'sentry/utils/reactRouter6Compat';
```
Comment (priscilawebdev) [inline static/app/components/links/link.tsx:8]: nit: are we sure about naming it `locationDescriptor**ToTo**` ?

File: static/app/utils/browserHistory.tsx [hunk 2]
```diff
@@ -14,4 +18,71 @@ import {browserHistory as react3BrowserHistory} from 'react-router';
...
+  // functions to keep things working
+  const compat6BrowserHistory: History = {
+    push: to => router.navigate(locationDescriptorToTo(to)),
+    replace: to => router.navigate(locationDescriptorToTo(to), {replace: true}),
+    go: n => router.navigate(n),
+    goBack: () => router.navigate(-1),
+    goForward: () => router.navigate(1),
+
+    // XXX(epurkhiser): react-router 6's BrowserHistory does not let you create
+    // multiple lsiteners. This implementation is similar but allows multiple
```
Comment (priscilawebdev) [inline static/app/utils/browserHistory.tsx:39]:
```suggestion
    // multiple listeners. This implementation is similar but allows multiple
```

**Comment by priscilawebdev** [comment]
- Are we planning a gradual migration?
- What are the blockers?
- I've heard that React Router 6 comes with many breaking changes. What steps are we taking to ensure we don't break anything?

It looks good so far and I am looking forward to it 🚀

**Review by scttcper — APPROVED** [review]
seems to work

**Pull request merged**